src/ameba/cli/cmd.cr

```diff
@@ -177,6 +182,30 @@ module Ameba::Cli
     config
   end
 
+  private def root_path_from_opts(opts)
+    return unless globs = opts.globs
+    root =
+      case
+      when path = opts.config
+        path.parent
+      when path = globs.find(&->File.file?(String))
+        Path[path].parents.reverse!.find(&->root_path?(Path))
+      when path = globs.find(&->File.directory?(String))
+        path = Path[path]
+        if root_path?(path)
+          path
+        else
+          path.parents.reverse!.find(&->root_path?(Path))
+        end
+      end
+    root.try &.expand(home: true)
+  end
+
+  private def root_path?(path)
+    File.exists?(path / Config::FILENAME) ||
+      File.exists?(path / "shard.yml")
+  end
```

Comment:
Could this mistakenly detect other shard directories? Should we read the contents of shard.yml to verify?

Reply:
In theory - yes, although I'd consider this rather an edge-case. You'd either point to a folder having it, or a subfolder. I don't really understand WDYM by _reading the contents of `shard.yml` to verify_ - verify _what_ exactly?

Reply:
I mistakenly thought we could read `shard.yml` and compare the shard name somehow.